tools/container/toolkit/toolkit.go

```diff
@@ -404,7 +417,65 @@ func Install(cli *cli.Context, opts *options) error {
 		log.Errorf("Ignoring error: %v", fmt.Errorf("error installing NVIDIA Container CDI Hook CLI: %v", err))
 	}
 
-	err = installToolkitConfig(cli, toolkitConfigPath, nvidiaContainerCliExecutable, nvidiaCTKPath, nvidiaContainerRuntimeHookPath, opts)
+	var runtimeBinaryPaths []string
+
+	if r, ok := os.LookupEnv("RUNTIME"); ok {
+
+		switch r {
+		case "containerd":
+			var cliArgs []string
+			if opts.HostRootMount != "" {
+				cliArgs = append(cliArgs, "chroot", opts.HostRootMount)
+			}
+			cliArgs = append(cliArgs, "containerd", "config", "dump")
+
+			cfg, err := containerd.New(
+				containerd.WithReference(pkgconfig.FromCLI(cliArgs)),
+			)
+			if err != nil {
+				return fmt.Errorf("unable to load containerd config: %w", err)
+			}
+
+			containerdDefaultRuntime := cfg.DefaultRuntime()
+			if containerdDefaultRuntime != "" {
+				runtimeData, err := cfg.GetRuntime(containerdDefaultRuntime)
+				if err == nil {
+					runtimeTOML := runtimeData.(*toml.Tree)
+					if binaryPath, ok := runtimeTOML.GetPath([]string{"options", "BinaryName"}).(string); ok && binaryPath != "" {
+						runtimeBinaryPaths = append(runtimeBinaryPaths, binaryPath)
+					}
+				}
+			}
```

Comment:
This definitely speaks to adding an interface that we return from `GetRuntime` instead.

Reply:
This is done